tests/drivers/dac/dac_api/boards/esp32s2_devkitc.overlay

```diff
@@ -0,0 +1,9 @@
+/*
+ * Copyright 2022 Espressif (Shanghai)
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+&dac {
+	status = "okay";
```

Comment:
As mentioned in the other PR, I prefer enabling the DAC in the board `.dts` file.

Reply:
Then code size would increase when dac isn't enabled in .dts:  

```c
config ADC_ESP32
	bool "ESP32 ADC driver"
	default y
	depends on DT_HAS_ESPRESSIF_ESP32_ADC_ENABLED
	help
	  Enable the driver implementation for the ESP32 ADC
```

Reply:
I guess you meant the DAC and not the ADC, but it's the same for both: The above snippet enables the specific DAC driver of that chip **only if** the DAC in general is enabled as well (by setting `CONFIG_DAC=y`). And in that case you do want to use the DAC and it's fine that the code size increases.

See here:

https://github.com/zephyrproject-rtos/zephyr/blob/de526632ae7a58eb323e68b7f44c76bd6a79c627/drivers/CMakeLists.txt#L30

https://github.com/zephyrproject-rtos/zephyr/blob/de526632ae7a58eb323e68b7f44c76bd6a79c627/drivers/dac/Kconfig#L14

https://github.com/zephyrproject-rtos/zephyr/blob/de526632ae7a58eb323e68b7f44c76bd6a79c627/drivers/dac/Kconfig#L54

Reply:
Sorry, DAC. Indeed, you are correct. Well, lots of overlay files could be removed after all, not only in this particular sample code.

Reply:
Possibly yes. But let's at least make sure that no further redundant ones are added :)

Reply:
I agree.

Reply:
@hakehuang Would you remove this esp32s2_devkitc overlay file and enable the dac into `esp32s2_devkitc.dts` instead?

Reply:
ok